### ISSUE TITLE
Empty word

### ISSUE THREAD

**Issue by kwalcock** [clulab/processors#630]
For the text below, the third sentence includes an empty word between "of" and "5,242".  Other parts of pipeline aren't expecting this.  It is happening in Tokenizer.readTokens already.  It could be filtered out right there, but it would also be good to know how it gets in there in the first place.

```
The multi-actor scientific committee (Research,

Wintering 2020/21 at the same period, i.e. a delay of 6,357 ha. The

development,

seedlings also show a downtrend for all

transformers ...), set up to take into

speculations, with a total area of ​​5,242 ha (cf. Table2).
```

**Comment by MihaiSurdeanu** [comment]
Nice catch! We did see PDF converters inserting white spaces that are not ASCII in their output. These Unicode white spaces need to be explicitly added to the tokenizer here:
https://github.com/clulab/processors/blob/master/main/src/main/java/org/clulab/processors/clu/tokenizer/OpenDomainLexer.g#L78

**Comment by kwalcock** [comment]
In this case it is a plain space in the text, but I can do testing with `&nbsp;` and similar characters as well.  @hubert10 ran into the problem with a file when I added an `assert(word.nonEmpty)`.  The internet is thinking there might be a rule in OpenDomainLexer.g that matches zero characters.

**Comment by kwalcock** [comment]
You are right.  Although the space is plain, there are some \u200B characters there as well, just to the right of it.  That's not in the list

```
WHITESPACE: (('\u0000'..'\u0020')|'\u2028'|'\u2029'|'\u0085'|'\u00A0'|('\u2000'..'\u200A')|'\u3000')+ -> skip ;
```

**Comment by kwalcock** [comment]
`antlr` is doing OK.  It's the ScienceUtils.replaceUnicodeWithAscii that is blanking out the word.  In `unicode_to_ascii.tsv` there is
```
# 200B-200F: various zero-width spaces and control chars
200B	
```
It is being called from TokenizerStepNormalization.process.  I'm checking for empty words there.

**Comment by kwalcock** [comment]
This is addressed with #631.
